cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -40,14 +40,16 @@ purge() {
     if [ "$distribution" = "ubuntu-14.04" ]; then
         # snap.mount.service is a trusty thing
         systemctl_stop snap.mount.service
-    else
-        # undo any bind mount to ${SNAP_MOUNT_DIR} that resulted from LP:#1668659
-        # (that bug can't happen in trusty -- and doing this would mess up snap.mount.service there)
-        if grep -q "${SNAP_MOUNT_DIR} ${SNAP_MOUNT_DIR}" /proc/self/mountinfo; then
-            umount -l "${SNAP_MOUNT_DIR}" || true
-        fi
     fi
 
+    # Undo any bind mounts to ${SNAP_MOUNT_DIR} or /var/snap done by parallel
+    # installs or LP:#1668659
+    for mp in "$SNAP_MOUNT_DIR" /var/snap; do
+        if grep -q "$mp $mp" /proc/self/mountinfo; then
+            umount -l "$mp" || true
+        fi
+    done
```

Comment:
This is being applied to trusty even though @mvo5 said we should ignore trusty. I'm not sure if that this is what @mvo5 meant. @mvo5 - can you comment?